pkg/operator/ceph/csi/util.go

```diff
@@ -132,13 +136,17 @@ func getToleration(provisioner bool) []corev1.Toleration {
 	return tolerations
 }
 
-func getNodeAffinity(provisioner bool) *corev1.NodeAffinity {
+func getNodeAffinity(clientset kubernetes.Interface, provisioner bool) *corev1.NodeAffinity {
 	// Add NodeAffinity if any
 	nodeAffinity := ""
+	var err error
 	if provisioner {
-		nodeAffinity = os.Getenv(provisionerNodeAffinityEnv)
+		nodeAffinity, err = k8sutil.GetOperatorSetting(clientset, provisionerNodeAffinityEnv, "")
 	} else {
-		nodeAffinity = os.Getenv(pluginNodeAffinityEnv)
+		nodeAffinity, err = k8sutil.GetOperatorSetting(clientset, pluginNodeAffinityEnv, "")
+	}
+	if err != nil {
+		logger.Warningf("node affinity will not be applied. %v", err)
```

Comment:
When there is an `err`, best practice is to return instead of continuing execution in the method even if `nodeAffinity != ""` is handled below. Then you likely don't even need to check for `nodeAffinity != ""`

Reply:
ack. fixed.